harness/determined/exec/harness.py

```diff
@@ -221,6 +221,11 @@ def main() -> None:
         f"New trial runner in (container {container_id}) on agent {agent_id}: {env.__dict__}."
     )
 
+
+    if env.experiment_config['checkpoint_storage']['type'] == 'azure':
+        from azure.storage.blob import BlobServiceClient
+        logging.getLogger("azure").setLevel(logging.ERROR)
```

Comment:
why aren't you doing that in `azure_client.py`, where `BlobServiceClient` is imported normally?

Reply:
I wasn't sure if it made sense to import azure_client here since its unused. I just wanted to import some code from the `azure` library so that the loggers would be created and I could set the level appropriately. I'm going to test [ryan's suggestion](https://github.com/determined-ai/determined/pull/2400#discussion_r646964058) which would resolve this as well.

Reply:
since you want to patch `azure`s logging config , and `azure_client.py` seems to be the only place where you import `azure`, and it won't log anything before it's imported, I think you can do this in `azure_client` right after the imports.